### PR TITLE
Story/fireweather bot/367

### PR DESCRIPTION
**Pull request by andrea-williams**
- Display noon forecast temp & RH in table & graph
- Forecast values displayed are pulled from most recently issued forecast

### CODE REVIEW

File: src/features/fireWeather/slices/forecastsSlice.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    getForecastsStart(state: State) {
+      state.loading = true
+    },
+    getForecastsFailed(state: State, action: PayloadAction<string>) {
+      state.loading = false
+      state.error = action.payload
+    },
+    getForecastsSuccess(state: State, action: PayloadAction<Forecast[]>) {
+      state.loading = false
+      state.error = null
```
Comment (Kyubinhan) [inline src/features/fireWeather/slices/forecastsSlice.ts:33]: Can you put these in at the end?
I realized processing data take times, and we want the loading spinner to keep going until the processing is done

File: src/features/fireWeather/slices/forecastsSlice.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        }
+      })
+      let counter = 0
+      for (const key in state.noonForecastsByStation) {
+        state.noonForecasts[counter] = {
+          station_code: key,
+          values: state.noonForecastsByStation[key]
+        }
+        counter++
+      }
```
Comment (Kyubinhan) [inline src/features/fireWeather/slices/forecastsSlice.ts:55]: I think we need to discuss about this. There must be a better way to process the data

File: src/features/fireWeather/components/DailyForecastDisplay.tsx [hunk 3]
```diff
@@ -23,67 +30,67 @@ const useStyles = makeStyles({
...
-interface Props {
-  values: ModelValue[] | undefined
-}
-
-const DailyModelsDisplay = ({ values }: Props) => {
+const DailyForecastDisplay = (props: {
+  values: (ModelValue | NoonForecastValue)[] | undefined
+  testId: string
+  title: string
+}) => {
```
Comment (Kyubinhan) [inline src/features/fireWeather/components/DailyForecastDisplay.tsx:37]: Can you define the `Props` interface instead of doing it here?

Comment (Kyubinhan) [reply src/features/fireWeather/components/DailyForecastDisplay.tsx:37]: And do the object destructuring assignment like `({ values, testId, title}: Props)`

File: src/features/fireWeather/components/TempRHGraph.styles.ts [hunk 4]
```diff
@@ -61,6 +61,16 @@ export const useStyles = makeStyles({
       stroke: '#03a1fc',
       fill: 'none',
       cursor: 'pointer'
+    },
+    '& .forecastTempDot': {
+      stroke: 'salmon',
+      fill: 'none',
+      cursor: 'pointer'
+    },
+    '& .forecastRhDot': {
```
Comment (Kyubinhan) [inline src/features/fireWeather/components/TempRHGraph.styles.ts:70]: forecastRhDot -> forecastRHDot

File: src/features/fireWeather/components/WxDataGraph.stories.tsx [hunk 5]
```diff
@@ -91,6 +92,71 @@ const getPastValues = () => {
...
+    title: 'Noon Forecast Values (UTC Timezone)'
+  }
+
+  return (
+    <>
+      <DailyWeatherDisplay values={modelValues} tableMetadata={modelTableMetadata} />
+      <DailyWeatherDisplay values={forecastValues} tableMetadata={forecastTableMetadata} />
+    </>
+  )
+})
```
Comment (Kyubinhan) [inline src/features/fireWeather/components/WxDataGraph.stories.tsx:159]: Can you create a new story file for `DailyWeatherDisplay`?

File: src/features/fireWeather/components/WxDataGraphToggles.tsx [hunk 6]
```diff
@@ -29,14 +32,18 @@ const WxDataToggles = ({
...
+  showForecasts,
+  setShowForecasts
 }: Props) => {
   const classes = useStyles()
 
   return (
     <FormGroup row>
       <FormControlLabel
         className={classes.formControlLabel}
+        data-testid="show-readings-toggle"
```
Comment (Kyubinhan) [inline src/features/fireWeather/components/WxDataGraphToggles.tsx:46]: These newly added 3 test ids are not being used anywhere, did you intend to test them?

File: src/features/fireWeather/components/TempRHGraph.styles.ts (hunk 4 above)
Comment (andrea-williams) [reply src/features/fireWeather/components/TempRHGraph.styles.ts]: Fixed

File: src/features/fireWeather/components/WxDataGraphToggles.tsx (hunk 6 above)
Comment (andrea-williams) [reply src/features/fireWeather/components/WxDataGraphToggles.tsx:46]: oops. that's leftover from a while ago when I was trying to make up more tests to get >80% coverage. I'll delete the IDs now - I'm not using them and I don't think they're needed anymore

File: src/features/fireWeather/components/WxDataGraph.stories.tsx (hunk 5 above)
Comment (andrea-williams) [reply src/features/fireWeather/components/WxDataGraph.stories.tsx]: Done

**Pull request merged**